inductiva/molecules/simulators/gromacs.py

```diff
@@ -16,50 +16,40 @@ def api_method_name(self) -> str:
 
     def run(
         self,
-        input_dir: types.Path,
-        sim_config_filename: str,
-        protein_filename: str,
-        topology_filename: str,
-        output_dir: Optional[types.Path] = None,
+        input_directory: types.Path,
+        method_name: str,
         track_logs: bool = False,
+        output_directory: Optional[types.Path] = None,
+        **gromacs_flags: Optional[types.Args],
     ) -> pathlib.Path:
-        """Run the energy minimization of a protein.
+        """Run a specified gromacs method.
 
         Args:
             input_dir: Path to the directory containing the input files.
-            sim_config_filename: Name of the .mdp file containing the
-                energy minimization parameters.
-            protein_filename: Name of the file containing the protein
-                .gro file.
-            topology_filename: Name of the file containing the topology
-                .top file.
+            method_name: Method to run.
+            output_dir: Path to the directory where the output files will be
+                stored.
+            gromacs_flags: Flags to pass to the gromacs CLI.
         """
-        return super().run(input_dir,
-                           output_dir=output_dir,
+        return super().run(input_directory,
+                           output_dir=output_directory,
                            track_logs=track_logs,
-                           input_filename=sim_config_filename,
-                           protein_filename=protein_filename,
-                           topology_filename=topology_filename)
+                           method=method_name,
+                           user_flags=gromacs_flags)
 
     def run_async(
         self,
         input_dir: types.Path,
-        sim_config_filename: str,
-        protein_filename: str,
-        topology_filename: str,
+        method_name: str,
+        **gromacs_flags: Optional[types.Args],
```

Comment:
Just a quick note: it seems a bit inconsistent to have one type called Args but then the name of the argument is gromacs_FLAGS. Maybe the type should be called Args (too generic?) but should be called CommandLineFlags.
But I am not sure about this. It just feels weird...

Reply:
Good catch, that was a type, should be str instead of types.Args